ComSemApp/teacher/views.py

```diff
@@ -355,3 +357,93 @@ def delete_file(url):
         os.remove(url)
     except FileNotFoundError:
         pass
+
+
+class SpeakingPracticeAttemptReviewView(TeacherCourseViewMixin, ListView):
+    model = SpeakingPracticeAttempt
+    template_name = "ComSemApp/teacher/attempt_request_review.html"
+
+    def get_context_data(self, **kwargs: any) -> dict[str, any]:
+
+        context = super().get_context_data(**kwargs)
+
+        return context
+
+class SpeakingPracticeReviewRequestsListView(TeacherCourseViewMixin, ListView):
+    model = SpeakingPracticeAttemptReviewRequest
+    template_name = 'ComSemApp/teacher/request_list.html'
+    context_object_name = 'requests'
+
+    def get_queryset(self):
+        requests_queryset = self.course.get_speaking_practice_review_requests().order_by('review','-date')
+
+        return requests_queryset
+
+
+class SpeakingPracticeAttemptReviewCreateView(TeacherCourseViewMixin, CreateView):
+    model = SpeakingPracticeAttemptReview
+    template_name = "ComSemApp/teacher/attempt_request_info.html"
+    fields = ["request", "reviewer", "comments"]
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        attempt_id = self.kwargs.get('attempt_id')
+        attempt = get_object_or_404(SpeakingPracticeAttempt, id=attempt_id)
+        context['attempt'] = attempt
+        return context
+    
+
+    def form_invalid(self, form):
+        response = super().form_invalid(form)
+        return JsonResponse(form.errors, status=400)
+
+    def form_valid(self, form):
+        review = form.save(commit=False)
+        newScore = self.request.POST.get('newScore')
+        id = self.request.POST.get('request')
+        attempt = get_object_or_404(SpeakingPracticeAttempt, id=id)
+
+        if newScore is not None or newScore is not "":
+            attempt.correct = newScore
+            attempt.save()
+            review.original_score = self.request.POST.get('originalScore')
```

Comment:
The post request should not be sending original score. The process for this should be as follows:
`if new_score:`
`  # get attempt`
`  review.original_score = attempt.correct`
`  attempt.correct = new_score`
`  attempt.save()`
`review.save()`